### ISSUE TITLE
Received notifications are duplicated

### ISSUE THREAD

**Issue by IgorRosocha** [firebase/firebase-ios-sdk#8282]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 12.5
  * Firebase SDK version: 8.1.1
  * Installation method: Carthage
  * Firebase Component: Messaging

#### Steps to reproduce:

Since using Firebase Messaging `7.11.0`, multiple users started to complain that after some time all of the notifications received from our apps are duplicated. We have currently received several reports from 2 of our production apps. Notifications are being sent using `Firebase Console` to relevant topics. Unfortunately, we were not able to reproduce this issue locally. Users also reported that issue got fixed by reinstalling the app.

However, we have noticed that in some cases, SDK sends multiple FCM tokens (4-6) in a row, f.e. right after reinstalling the app. Could the problem be related to incorrect behavior of SDK, leaving user with multiple valid FCM tokens? Has anyone experienced the same problem?

**Comment by chliangGoogle** [comment]
@IgorRosocha Sending multiple FCM tokens (4-6) is definitely abnormal behavior. Although 7.11.0 is separating token logics between InstanceiD and messaging SDKs, so I suspect there might be some race conditions handling tokens between InstanceID and Messaging (even though we've tested that it shouldn't).

Our latest version no longer has InstanceID that should for sure resolve the suspicious condition above, would you mind updating to the latest SDK and remove InstanceID to try it out? The migration guide is [here](https://firebase.google.com/docs/projects/manage-installations#fid-iid)

**Comment by IgorRosocha** [comment]
@chliangGoogle thanks for you response! 

> Our latest version no longer has InstanceID that should for sure resolve the suspicious condition above, would you mind updating to the latest SDK and remove InstanceID to try it out? The migration guide is here

As mentioned above, first time user has reported the issue was from the version 7.11.0, but the latest report was actually from the latest 8.1.1 version of the SDK. We are using Carthage distribution and not calling InstanceID directly. I have noticed that we had InstanceID included in frameworks, so I have removed it. However, after launching the app again, I have immediately received 4 different FCM tokens in a row. However, I am not sure if this is the cause of the issue - when I tried sending notifications, they were delivered correctly.
